cumulus_library/studies/discovery/discovery_templates/code_system_pairs.sql.jinja

```diff
@@ -0,0 +1,44 @@
+{%- import 'syntax.sql.jinja' as syntax -%}
+
+{%- macro get_source_col(column_hierarchy) -%}
+{%- if column_hierarchy|length >1 or column_hierarchy[0][1].__name__ == 'list'-%}
+table_{{ column_hierarchy|length  }}.col_{{ column_hierarchy|length  }}
+{%- else -%}
+{{ column_hierarchy[0][0] }}
+{%-endif %}
+{%- endmacro -%}
+
+CREATE TABLE {{ output_table_name }} AS
+{%- for table in code_system_tables %}
+{%- if table.has_data %}
+SELECT DISTINCT
+    '{{ table.table_name }}' AS table_name,
+    '{{ table.column_display_name }}' AS column_name,
+    {{ get_source_col(table.column_hierarchy) }}.code,
+    {{ get_source_col(table.column_hierarchy) }}.display,
+    {{ get_source_col(table.column_hierarchy) }}.system
+FROM {{ table.table_name }}
+{%- if table.column_hierarchy|length >1 or table.column_hierarchy[0][1].__name__ == 'list' -%},
+UNNEST({{ table.column_hierarchy[0][0] }}) AS table_1 (col_1)
+{%- endif %}
+{%- for index in range(1,table.column_hierarchy|length) %}
+{%- if table.column_hierarchy[index][1].__name__ == 'list' -%},
+UNNEST(col_{{index}}.{{ table.column_hierarchy[index][0].split('.')[0] }}) as table_{{ index +1 }} (col_{{ index +1 }})
```

Comment:
This confused me a bit (thought there was a bug) until I got to the `.py` file and read your whole squashing thing.

Maybe a comment above this block that says "Because the hierarchy is coming in squashed by the Python code, we can assume that all hierarchy levels except the last will be a list. The last could be a list or dict." but prettier.

Not necessary, but who knows in what order a future dev will be reading these, and that unspoken assumption is a critical component of this logic.

Reply:
yeah that's fair - i'll add that.